Sources/SwiftKafka/KafkaConsumer.swift

```diff
@@ -0,0 +1,418 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import struct Foundation.UUID // TODO: can we avoid this Foundation import??
+import Logging
+import NIOCore
+
+/// `NIOAsyncSequenceProducerDelegate` implementation handling backpressure for ``KafkaConsumer``.
+actor ConsumerMessagesAsyncSequenceDelegate: NIOAsyncSequenceProducerDelegate {
```

Comment:
Went with an `actor` as we need to be thread-safe and have this `weak var consumer`. However, we could think about a different approach here.